### PR TITLE
Refactor/convention plugins

### LINKED ISSUE #135: [refactor] plugin convention

### PR DESCRIPTION
**Pull request by Kaaveh**
close #135

### CODE REVIEW

File: build-logic/convention/build.gradle.kts [hunk 1]
```diff
@@ -41,40 +25,40 @@ gradlePlugin {
     plugins {
         register("androidApplication") {
             id = "composenews.android.application"
-            implementationClass = "AndroidApplicationConventionPlugin"
+            implementationClass = "ApplicationConventionPlugin"
```
Comment (VahidGarousi) [inline build-logic/convention/build.gradle.kts:28]: Why did you remove the Android prefix?
this plugin is for Android projects; in my opinion, we can create other plugins for iOS-related projects later.
so I can't entirely agree with this change

File: build-logic/convention/src/main/kotlin/ir/composenews/Compose.kt [hunk 2]
```diff
@@ -29,9 +28,9 @@ internal fun Project.configureAndroidCompose(
         dependencies {
             val bom = libs.findLibrary("compose-bom").get()
             val composeCompiler = libs.findLibrary("compose-compiler").get()
-            add("implementation", platform(bom))
-            add("implementation", composeCompiler)
-            add("androidTestImplementation", platform(bom))
```
Comment (VahidGarousi) [inline build-logic/convention/src/main/kotlin/ir/composenews/Compose.kt:34]: Why you remove `add`?

Comment (Kaaveh) [reply build-logic/convention/src/main/kotlin/ir/composenews/Compose.kt:34]: I checked and saw this syntax works:

```kotlin
"implementation"(composeCompiler)
"androidTestImplementation"(platform(bom))
```

File: build-logic/convention/build.gradle.kts (hunk 1 above)
Comment (Kaaveh) [reply build-logic/convention/build.gradle.kts:28]: I agree with you. I'll change it.

**Review by VahidGarousi — APPROVED** [review]

**Pull request merged**